VenafiPS/Public/Test-TppToken.ps1

```diff
@@ -0,0 +1,170 @@
+
+.PARAMETER TppToken
+Token object obtained from New-TppToken
+
+.PARAMETER VenafiSession
+Session object created from New-VenafiSession method.  The value defaults to the script session object $VenafiSession.
+
+.PARAMETER GrantDetail
+Provides detailed info about the token object from the TPP server response as an output.
+PSCustomObject with the following properties:
+    AuthUrl
+    AccessToken
+    RefreshToken
+    Scope
+    Identity
+    TokenType
+    ClientId
+    Expires
+
+.INPUTS
+Accesstoken
+
+.OUTPUTS
+Boolean (default).
+
+.EXAMPLE
+Test-TppToken
+Verify that accesstoken stored in $VenafiSession object is valid.
+
+.EXAMPLE
+$TppToken | Test-TppToken
+Verify that token object from pipeline is valid. Can be used to validate directly object from New-TppToken.
+
+.EXAMPLE
+Test-TppToken -AuthServer 'mytppserver.example.com' -AccessToken $cred
+Verify that PsCredential object containing accesstoken is valid.
+
+.EXAMPLE
+Test-TppToken -GrantDetail
+Verify that accesstoken stored in $VenafiSession object is valid and return PsCustomObject as output with details.
+
+.LINK
+http://VenafiPS.readthedocs.io/en/latest/functions/Test-TppToken/
+
+.LINK
+https://github.com/gdbarron/VenafiPS/blob/main/VenafiPS/Code/Public/Test-TppToken.ps1
+
+.LINK
+https://docs.venafi.com/Docs/20.4SDK/TopNav/Content/SDK/AuthSDK/r-SDKa-GET-Authorize-Verify.php?tocpath=Auth%20SDK%20reference%20for%20token%20management%7C_____13
+
+#>
+function Test-TppToken {
+
+    [CmdletBinding(DefaultParameterSetName = 'Session')]
+    param (
+        [Parameter(Mandatory, ParameterSetName = 'AccessToken')]
+        [ValidateScript( {
+                if ( $_ -match '^(https?:\/\/)?(((?!-))(xn--|_{1,1})?[a-z0-9-]{0,61}[a-z0-9]{1,1}\.)*(xn--)?([a-z0-9][a-z0-9\-]{0,60}|[a-z0-9-]{1,30}\.[a-z]{2,})$' ) {
+                    $true
+                } else {
+                    throw 'Please enter a valid server, https://venafi.company.com or venafi.company.com'
+                }
+            }
+        )]
+        [Alias('Server')]
+        [string] $AuthServer,
+
+        [Parameter(Mandatory, ParameterSetName = 'AccessToken', ValueFromPipeline)]
+        [PSCredential] $AccessToken,
+
+        [Parameter(Mandatory, ParameterSetName = 'TppToken', ValueFromPipeline)]
+        [pscustomobject] $TppToken,
+
+        [Parameter()]
+        [switch] $GrantDetail,
+
+        [Parameter(ParameterSetName = 'Session')]
+        [VenafiSession] $VenafiSession = $script:VenafiSession
+    )
+
+    begin {
+        $params = @{
+            Method  = 'Get'
+            UriRoot = 'vedauth'
+            UriLeaf = 'Authorize/Verify'
+        }
+    }
+
+    process {
+        
+        Write-Verbose ('Parameter set: {0}' -f $PSCmdlet.ParameterSetName)
+
+        switch ($PsCmdlet.ParameterSetName) {
+            'Session' {
+                $VenafiSession.Validate('token') | Out-Null
```

Comment:
we are going to validate in this function so this line isn't needed.  and as of right now, it doesn't validate the token is actually valid...yet

Reply:
Removed